src/main/java/spark/Response.java

```diff
@@ -67,12 +69,37 @@ public void type(String contentType) {
     }
 
     /**
+     * Sets the content length for the response
+     *
+     * @param contentLength the content length
+     */
+    public void length(int contentLength) {
+        response.setContentLength(contentLength);
+
+        contentLengthManuallySet = true;
+    }
+
+    /**
      * Sets the body
      *
      * @param body the body
      */
     public void body(String body) {
         this.body = body;
+
+        if (!contentLengthManuallySet) {
+            String charsetValue = response.getCharacterEncoding() == null ?
+                    "ISO-8859-1" // recommended charset according to http://www.ietf.org/rfc/rfc2047.txt
+                    : response.getCharacterEncoding();
+
+            Charset charset = null; //Charset.forName(charsetValue);
+
+            if (charset != null) {
+                response.setContentLength(this.body.getBytes(charset).length);
+            } else if (LOG.isDebugEnabled()) {
```

Comment:
im not 100% sure abut this?, isnt those new loggers do this already. Do you have to write a if condition to check debug ?. Any way your string is passed with {}. so unless the logger get activated it should not do string operations